File: src/main/java/me/ctknight/uploadmanager/UploadException.java
/*
 * Copyright (c) 2016. All rights reserved. Lai Jiewen <alanljw12345@gmail.com>
 */

package me.ctknight.uploadmanager;

public class UploadException extends Exception {

    public UploadException() {}

    public UploadException(String detailMessage) {
        super(detailMessage);
    }

    public UploadException(String detailMessage, Throwable throwable) {
        super(detailMessage, throwable);
    }

    public UploadException(Throwable throwable) {
        super(throwable);
    }
}

File: lib/src/main/java/me/ctknight/uploadmanager/UploadException.kt
/*
 * Copyright (c) 2018. All rights reserved. Lai Jiewen <alanljw12345@gmail.com>
 */

package me.ctknight.uploadmanager

open class UploadException : Exception {

  internal constructor()

  internal constructor(detailMessage: String) : super(detailMessage)

  internal constructor(detailMessage: String, throwable: Throwable) : super(detailMessage, throwable)

  internal constructor(throwable: Throwable) : super(throwable)
}
